### DISCUSSION TITLE
RTL and LTR at the same time

### DISCUSSION THREAD

**Discussion by itismohammadreza** [twbs/bootstrap#37877] [Q&A]
Please someone explain to me that if we use the solution that Bootstrap explained in [documentation](https://getbootstrap.com/docs/5.0/getting-started/rtl/#ltr-and-rtl-at-the-same-time) to use RTL and LTR at the same time, is it supposed to generate a new file for RTL version ? Or do all these events happen in one file?
If a new file is generated, this file is also available for download on the Bootstrap site itself.

Please tell me step by step how I should implement a bilingual site with this solution ?
In simpler terms, I want in RTL version `ms-2` mean `margin-right` and in LTR version it mean `margin-left`.

**Comment by denizgolbas** [comment]
Yes. Instead of left or right you need to use start and end. 
For LTR version start is left , end is right.
When you change language if the language is RTL it will automaticly seen reversed.

**Comment by itismohammadreza** [comment]
@denizgolbas 
Thanks, but when I change language It is not effective and still e.g. `ms` means `margin-left`.

**Reply by denizgolbas** [reply]
You need to add dir attiributte. When the user choose the RTL language. You need to make dynamically with your backend or frontend part. This manupulation process up to you.
```@html
<html lang="fa" dir="rtl">
```

**Comment by ffoodd** [comment]
[The current way RTL Bootstrap is built relies on RTLCSS](https://getbootstrap.com/docs/5.0/getting-started/rtl/#approach) and is meant to ease the switch: you can use the exact same markup (except the `lang` and `dir` attributes, obviously) and only switch from `bootstrap.min.css` to `bootstrap.rtl.min.css` for RTL languages.

When using the snippet to use both directions at the same time, you need to build you CSS by yourself:
1. use the provided snippet, 
2. then compile with Sass to get the `.ltr` wrapped file (expanded Sass styles to keep RTLCSS comments),
3. and finally run [RTLCSS](https://rtlcss.com/) on the outputted file to get the `.rtl` wrapped file.

You should get two different files, that you may merge and minify using Sass or anything else.

Then you would need to use `.ltr` and `.rtl` classes where you need to change the direction. The snippet can be adapted to use `[dir]` attribute instead of classes, but RTLCSS configuration is easier with classes.

This is quite laborious, but that's not a case we aimed to handle when we started the RTL support.

I hope this help!
